### PR TITLE
Convert symbol-search crash to NFW

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1584512

### CODE REVIEW

**Review by arkalyanms — APPROVED** [review]

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs [hunk 1]
```diff
@@ -554,28 +553,23 @@ private async Task<XElement> TryDownloadFileAsync(IFileDownloader fileDownloader
...
-                    return result;
-                }
-                catch (Exception e) when (ReportAndDoNotCatch(e))
-                {
-                    throw ExceptionUtilities.Unreachable();
+                    return (element, delay: default);
                 }
-
-                bool ReportAndDoNotCatch(Exception exception)
+                catch (Exception e) when (_service._reportAndSwallowExceptionUnlessCanceled(e, cancellationToken))
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:566]: What's with the _service. thing here having a function we're calling as opposed to just using the existing static methods?

**Review by jasonmalinowski — APPROVED** [review]

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:566]: (this is not blocking for taking this in to 17.4 of course)

Comment (dibarbet) [reply src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:566]: same q

**Review by dibarbet — APPROVED** [review]

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:566]: this was for testing purposes.  we actually validate these codepaths in tests, and it allows the test system to inject a helper that fails-fast as opposed to gracefully falling over (like the main code does).

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs [hunk 2]
```diff
@@ -511,23 +511,22 @@ private async Task<XElement> DownloadFileAsync(string serverPath, CancellationTo
...
                         // File was downloaded.  
-                        return resultOpt;
+                        return element;
                     }
                 }
             }
 
             /// <summary>Returns 'null' if download is not available and caller should keep polling.</summary>
-            private async Task<XElement> TryDownloadFileAsync(IFileDownloader fileDownloader, CancellationToken cancellationToken)
+            private async Task<(XElement? element, TimeSpan delay)> TryDownloadFileAsync(IFileDownloader fileDownloader, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs:529]:
```suggestion
            private async Task<(XElement element, TimeSpan delay)> TryDownloadFileAsync(IFileDownloader fileDownloader, CancellationToken cancellationToken)
```

**Pull request merged**